plugin/src/compilation/decompiler.ml

```diff
@@ -6,14 +6,18 @@ open Envutils
 open Tactics
 open Pp
 open Contextutils
-
+open Equtils
+open Apputils
+   
 (* Abstraction of Coq tactics supported by this decompiler.
    Serves as an intermediate representation that can be either
    transformed into a string or a sequence of actual tactics. *)
 type tact =
   | Intro of Id.t
   | Intros of Id.t list
   | Apply of env * types
+  (* Proof that x = y if true, y = x if false. *)
+  | Rewrite of env * types * bool
```

Comment:
Having an IR here seems really useful if you need to change details later. Good call earlier